### PR TITLE
Make eltype compile time viable

### PR DESCRIPTION
**Pull request by oxinabox**

### CODE REVIEW

**Comment by ararslan** [comment]
Implementation looks good, but some tests would be nice.

**Comment by oxinabox** [comment]
TBH: I have basically no idea what a `DispatchNode` is, 
and only  the vaguest idea of what this package does.

The `eltype` line will be hit by any existing tests, that hit it before.
Which may have been none.

File: src/nodes.jl [hunk 1]
```diff
@@ -678,10 +678,10 @@ else
     function Base.iterate(node::DispatchNode, state::Int=1)
         return IndexNode(node, state), state + 1
     end
+    Base.IteratorEltype(::Type{<:DispatchNode}) = Base.HasEltype()
```
Comment (iamed2) [inline src/nodes.jl]: This is already the default

**Comment by iamed2** [comment]
This doesn't change anything; `HasEltype` is the default.

**Comment by oxinabox** [comment]
See line 684, which is the actual change?

**Comment by iamed2** [comment]
Ah I see, sorry. Yeah that line seems reasonable. No need for the first one though.

**Pull request merged**